### PR TITLE
Add support to Common Boards to STM32F7 and ProjectLab board

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add support to Common Boards to STM32F7 and ProjectLab board
## Impact
Meadow users will be able to use ProjectLab with NuttX
## Testing
ProjectLab board

### CODE REVIEW

**Comment by acassis** [comment]
Hmm, seems like add common board support will requires all boards to have Make.defs instead Makefile:
```
====================================================================================
Configuration/Tool: stm32f746g-disco/lvgl,CONFIG_ARM_TOOLCHAIN_CLANG
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
  Building NuttX...
Makefile:23: board/Make.defs: No such file or directory
make[2]: *** No rule to make target 'board/Make.defs'.
make[2]: Failed to remake makefile 'board/Make.defs'.
make[1]: *** [Makefile:79: context] Error 2
make: *** [tools/Unix.mk:443: boards/.context] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 370: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
```

**Comment by raiden00pl** [comment]
> Hmm, seems like add common board support will requires all boards to have Make.defs instead Makefile:

That's true. The advantage is that it forces some standardization for boards build system :)

**Comment by acassis** [comment]
Our Mac CI fails very often, any idea?

```
==> Upgrading python@3.11
  3.11.4_1 -> 3.11.5 

==> Pouring python@3.11--3.11.5.ventura.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/2to3
Target /usr/local/bin/2to3
already exists. You may want to remove it:
  rm '/usr/local/bin/2to3'

To force the link and overwrite all conflicting files:
  brew link --overwrite python@3.11

To list all files that would be deleted:
  brew link --overwrite --dry-run python@3.11
```

**Comment by btashton** [comment]
> Our Mac CI fails very often, any idea?
> 
> ```
> ==> Upgrading python@3.11
>   3.11.4_1 -> 3.11.5 
> 
> ==> Pouring python@3.11--3.11.5.ventura.bottle.tar.gz
> Error: The `brew link` step did not complete successfully
> The formula built, but is not symlinked into /usr/local
> Could not symlink bin/2to3
> Target /usr/local/bin/2to3
> already exists. You may want to remove it:
>   rm '/usr/local/bin/2to3'
> 
> To force the link and overwrite all conflicting files:
>   brew link --overwrite python@3.11
> 
> To list all files that would be deleted:
>   brew link --overwrite --dry-run python@3.11
> ```

Because python is being upgraded we may need to first remove these existing symlinks. There are a couple more iirc. We removed them when they were not required for the AVR workaround but may need to bring them back. 

Similar issue to the openssl one fixed here 
https://github.com/apache/nuttx/pull/10116

**Comment by pkarashchenko** [comment]
> > Our Mac CI fails very often, any idea?
> > 
> > ```
> > ==> Upgrading python@3.11
> >   3.11.4_1 -> 3.11.5 
> > 
> > ==> Pouring python@3.11--3.11.5.ventura.bottle.tar.gz
> > Error: The `brew link` step did not complete successfully
> > The formula built, but is not symlinked into /usr/local
> > Could not symlink bin/2to3
> > Target /usr/local/bin/2to3
> > already exists. You may want to remove it:
> >   rm '/usr/local/bin/2to3'
> > 
> > To force the link and overwrite all conflicting files:
> >   brew link --overwrite python@3.11
> > 
> > To list all files that would be deleted:
> >   brew link --overwrite --dry-run python@3.11
> > ```
> 
> Because python is being upgraded we may need to first remove these existing symlinks. There are a couple more iirc. We removed them when they were not required for the AVR workaround https://github.com/apache/nuttx/pull/9669 but may need to bring them back. 
> 
> Similar issue to the openssl one fixed here 
> https://github.com/apache/nuttx/pull/10116
> 
> @xiaoxiang781216 and @pkarashchenko  since you both have delt with this issue in the past 
> 
> 

Will try to take a look today

**Comment by acassis** [comment]
Thank you @btashton and @pkarashchenko ! Finally it passed in the CI

File: boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_bringup.c [hunk 1]
```diff
@@ -56,6 +114,10 @@ int stm32_bringup(void)
 {
   int ret = OK;
 
+#if defined(CONFIG_I2C) && defined(CONFIG_SYSTEM_I2CTOOL)
+  stm32_i2ctool();
+#endif
```
Comment (pkarashchenko) [inline boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_bringup.c:119]:
```suggestion
  stm32_i2ctool();
```

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**